hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -477,4 +478,43 @@ public void setActiveTimeline(HoodieActiveTimeline activeTimeline) {
   public void setTableConfig(HoodieTableConfig tableConfig) {
     this.tableConfig = tableConfig;
   }
+
+  //------------------------------------------------------
+  //        Resource initialization and cleanup
+  //------------------------------------------------------
+
+  private void init(Configuration conf, String basePath,
+      boolean loadActiveTimelineOnLoad, ConsistencyGuardConfig consistencyGuardConfig) {
+    log.info("Loading HoodieTableMetaClient from " + basePath);
+    this.basePath = basePath;
+    this.consistencyGuardConfig = consistencyGuardConfig;
+    this.hadoopConf = new SerializableConfiguration(conf);
+    Path basePathDir = new Path(this.basePath);
+    this.metaPath = new Path(basePath, METAFOLDER_NAME).toString();
+    Path metaPathDir = new Path(this.metaPath);
+    this.fs = getFs();
+    DatasetNotFoundException.checkValidDataset(fs, basePathDir, metaPathDir);
+    this.tableConfig = new HoodieTableConfig(fs, metaPath);
+    this.tableType = tableConfig.getTableType();
+    log.info("Finished Loading Table of type " + tableType + " from " + basePath);
+    if (loadActiveTimelineOnLoad) {
+      log.info("Loading Active commit timeline for " + basePath);
+      getActiveTimeline();
+    }
+  }
+
+  private void cleanup() throws IOException {
+    this.basePath = null;
+    this.consistencyGuardConfig = null;
+    this.hadoopConf = null;
+    this.metaPath = null;
+    if (this.fs != null) {
```

Comment:
ah.. lets not close the fs object... This instance will be cached at the FileSystem level and closing this here, would make other threads (e.g with num.cores > 1, other spark executor's thread on same jvm to fail). #620 we have worked through this before..

Reply:
I just see the source code of Hadoop `FileSystem`. You are right. `FileSystem.get` will get an instance from cache, `FileSystem.newInstance` can returns a uniquie File System object. Sorry, I did not pay attention to it.